packages/app/hooks/useEntryPrice.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+import { getEntryPrice } from "app/api/rest/entryPrice";
+
+const savedPrice: Map<string, number> = new Map();
+
+const getIdentifier = (code?: string | null, issuer?: string | null) =>
+  code && issuer ? `${code}-${issuer}` : "";
+
+export default function useEntryPrice(
+  code?: string | null,
+  issuer?: string | null
+) {
+  const identifier = getIdentifier(code, issuer);
+  const [data, setData] = useState(savedPrice.get(identifier));
+
+  useEffect(() => {
+    if (data === undefined && code && issuer) {
+      fetchData(code, issuer).then(setData);
```

Comment:
```suggestion
      const fetchedData = await fetchData(code, issuer);
      if (fetchedData) {
      setData(fetchedData);
      }
```

Reply:
we should stick to `async/await` syntax

Reply:
Ok, will fix it, but the fix is not that simple. we can't use await in use effect

Reply:
You can simply create an async function and call it at the end of the `useEffect`.